swarm/api/http/server.go

```diff
@@ -1011,6 +1012,8 @@ func (b bufferedReadSeeker) Seek(offset int64, whence int) (int64, error) {
 }
 
 func (s *Server) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	ctx := context.TODO()
+
 	defer metrics.GetOrRegisterResettingTimer(fmt.Sprintf("http.request.%s.time", r.Method), nil).UpdateSince(time.Now())
 	req := &Request{Request: *r, ruid: uuid.New()[:8]}
```

Comment:
Note that even if we don't go through with Opentracing, at least `ruid` belongs to the context and not to the `Request` obj. We would be able to propagate it much further with context and get more meaningful traces through this correlation id.